code/go/0chain.net/smartcontract/storagesc/test_blobber_reward.go

```
@@ -0,0 +1,183 @@
+package storagesc
+
+import (
+	"0chain.net/core/common"
+	"0chain.net/smartcontract/stakepool/spenum"
+	"net/http"
+)
+
+func (srh *StorageRestHandler) getAllChallenges(w http.ResponseWriter, r *http.Request) {
+	// read all data from challenges table and return
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	allocationID := r.URL.Query().Get("allocation_id")
+
+	challenges, _ := edb.GetAllChallengesByAllocationID(allocationID)
```

Do not ignore errors